enigma-principal/enclave/src/epoch_keeper_t/nested_encoding.rs

```diff
@@ -0,0 +1,58 @@
+use std::vec::Vec;
+use ethereum_types::{U256,H160};
+
+pub const ONE: u8 = 1;
+pub const ZERO: u8 = 0;
+/// implements the serialization for types needed for epoch encoding in the KM node,
+/// according to this proof: https://github.com/enigmampc/protocol-discovery/blob/master/docs/hash_mul_nested.pdf
+pub trait NestedSerialization {
+    fn hash_encode(&self) -> (usize, Vec<u8>);
```

Comment:
This function is not implemented for the most basic type: &[u8]. Most of the types will be converted into that type.

Reply:
Currently, we decided to create this only for the epoch structure. in the future, we will add this as well.